roles/mysql/tasks/main.yml

```diff
@@ -0,0 +1,199 @@
+---
+# tasks file for mysql
+#
+- name: MySQL | configure apt repositories
+  ansible.builtin.apt_key:
+    keyserver: keyserver.ubuntu.com
+    id: "0xF1656F24C74CD1D8"
+
+- name: MySQL | add apt repositories
+  ansible.builtin.apt_repository:
+    repo: deb [arch=amd64] https://mirrors.gigenet.com/mariadb/repo/10.2/ubuntu {{ ansible_distribution_release }} main
+    state: present
+    update_cache: true
+
+- name: MySQL | Install database client packages
+  apt:
+    name: "{{ mysql_client_packages }}"
+    state: present
+    install_recommends: false
+    update_cache: true
+
+- name: MySQL | Install python database client packages
+  pip:
+    name: mysqlclient
+  when:
+    - running_on_server
+
+- name: MySQL | install mysql
+  ansible.builtin.apt:
+    name: "{{ mysql_packages }}"
+    state: present
+    update_cache: true
+  when:
+    - mysql_server
+  notify:
+    - set root password
+    - systemctl daemon-reexec
+
+
+- name: MySQL | configure mysql server
+  community.general.ini_file:
+    dest: "{{ mysql_configuration_destination }}/{{ mysql_name }}-server.cnf"
+    section: "{{ item.section }}"
+    option: "{{ item.option }}"
+    value: "{{ item.value }}"
+    mode: "0644"
+  loop: "{{ mysql_configuration_options }}"
+  loop_control:
+    label: "{{ item.option }}"
+  when:
+    - mysql_server
+  notify:
+    - restart mysql server
+
+- name: MySQL | configure mysql client
+  community.general.ini_file:
+    dest: "{{ mysql_configuration_destination }}/{{ mysql_name }}-client.cnf"
+    section: client
+    option: socket
+    value: "{{ mysql_socket }}"
+    mode: "0644"
+  when:
+    - mysql_server
```

Comment:
Do we want to restrict this task? We install the client packages even if `mysql_server` isn't set. Do we also want to configure the client in that case?

Reply:
yeah. This is how the db server will treat client connections to it.